python/hail/representation/annotations.py

```diff
@@ -47,3 +47,6 @@ def __repr__(self):
 
     def __eq__(self, other):
         return self._attrs == other._attrs
+
+    def __hash__(self):
+        return 37 + hash(tuple(sorted(self._attrs.items())))
```

Comment:
The hashes came up as an issue in one of the tests I made here.  Too much of a pain to split out every little thing!

Hash functions avoid collisions by adding multiples of primes.  I can't just return the hash of the tuples, because then there's no hash difference between a Struct and the tuple list.  @cseed second opinion?  I realize this is a bit hacky, there is probably a more standard way.